### DISCUSSION TITLE
Homebrew Cask

### DISCUSSION THREAD

**Discussion by iMonZ** [spacedriveapp/spacedrive#1547] [Feature Request]
Homebrew cask for macOS
brew.sh to automatically install and update this tool securly

**Comment by brxken128** [comment]
We plan to distribute Spacedrive to as many package repositories as we viably can - I do however think `brew` requires you to list your dependencies and provide a binary, however we distribute a fully qualified `.dmg` installer file which automatically updates itself on MacOS. It's not common to install GUI apps via brew, either - more often than not it's reserved for system libraries or command-line tools.

**Reply by iMonZ** [reply]
That’s extremely wrong.
Brew cask are for apps like Firefox, edge, chrome, ms teams etc.
Also yes for more technical people, brew is actually a very common way of installing.

**Reply by muescha** [reply]
Brew cask is particularly useful for GUI applications that come with .dmg packages, as it simplifies the installation process and eliminates the need for repetitive "drag and drop to Applications" actions.

The main question is whether Spacedrive will establish its own repository, known as a 'tap', where cask formulas can be updated through your continuous integration and continuous deployment (CI/CD) process. Alternatively, will it rely on users to provide the cask formula in the repository found at https://github.com/Homebrew/homebrew-cask.

For reference:
- You can find a comprehensive list of GUI apps installable from the internal cask repository here: https://formulae.brew.sh/cask/
- Instructions on how to write a cask formula can be found here: https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md

**Reply by iMonZ** [reply]
Please don’t use an own tab.
There is a option available in Homebrew Casks to automatically update when there is a new release in GitHub 
It’s called livecheck

**Reply by muescha** [reply]
but this update need to be done from a user? 

a tap updated by ci/cd tap is always up-to-date

and yes if it live in the cask repo then it is installable without tapping a new repo

**Reply by iMonZ** [reply]
No as I know it's actually not needed

**Reply by brxken128** [reply]
> Brew cask is particularly useful for GUI applications that come with .dmg packages, as it simplifies the installation process and eliminates the need for repetitive "drag and drop to Applications" actions.
> 
> The main question is whether Spacedrive will establish its own repository, known as a 'tap', where cask formulas can be updated through your continuous integration and continuous deployment (CI/CD) process. Alternatively, will it rely on users to provide the cask formula in the repository found at https://github.com/Homebrew/homebrew-cask.
> 
> For reference:
> - You can find a comprehensive list of GUI apps installable from the internal cask repository here: https://formulae.brew.sh/cask/
> - Instructions on how to write a cask formula can be found here: https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md

Thanks! I didn't realise GUI apps were so prevalent, I'll discuss this with the team and see what we can do.

**Reply by muescha** [reply]
there is no "auto update" for cask formula - a maintainer / user need to run livecheck or brew bump to create a new version and then push the updated cask to the repo and the PR need to be accepted.

**Reply by muescha** [reply]
@brxken128 FYI:  there is already a cask created: https://formulae.brew.sh/cask/spacedrive
:)


```bash
brew install spacedrive
==> Downloading https://github.com/spacedriveapp/spacedrive/releases/download/0.1.1/Spacedrive-darwin
==> Downloading from https://objects.githubusercontent.com/github-production-release-asset-2e65be/410
############################################################################################## 100.0%
==> Installing Cask spacedrive
==> Moving App 'Spacedrive.app' to '/Applications/Spacedrive.app'
🍺  spacedrive was successfully installed!
```

**Comment by muescha** [comment]
FYI:  there is already a cask created: https://formulae.brew.sh/cask/spacedrive

**Reply by Brendonovich** [reply]
We'll probably run `brew bump` as part of the release workflow to automate this
